internal/cloud/cloud_status_manager.go

```diff
@@ -10,28 +10,47 @@ import (
 	"sync"
 	"time"
 
+	"github.com/jonboulle/clockwork"
+
 	"github.com/windmilleng/tilt/internal/cloud/cloudurl"
 	"github.com/windmilleng/tilt/internal/feature"
 	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/token"
 	"github.com/windmilleng/tilt/pkg/logger"
+	"github.com/windmilleng/tilt/pkg/model"
 )
 
 // to avoid infinitely resubmitting requests on error
 const timeoutAfterError = 5 * time.Minute
 
+// how frequently we'll refresh cloud status, even if nothing changes
+const refreshPeriod = time.Hour
+
 const TiltTokenHeaderName = "X-Tilt-Token"
 const TiltTeamIDNameHeaderName = "X-Tilt-TeamID"
 
-func NewStatusManager(client HttpClient) *CloudStatusManager {
-	return &CloudStatusManager{client: client}
+func NewStatusManager(client HttpClient, clock clockwork.Clock) *CloudStatusManager {
+	return &CloudStatusManager{client: client, clock: clock}
+}
+
+// if any of these fields change, we know we need to do a fresh lookup
+type statusRequestKey struct {
+	tiltToken token.Token
+	teamID    string
+	version   model.TiltBuild
 }
 
 type CloudStatusManager struct {
 	client HttpClient
+	clock  clockwork.Clock
+
+	mu sync.Mutex
 
 	sleepingAfterErrorUntil time.Time
 	currentlyMakingRequest  bool
-	mu                      sync.Mutex
+
+	lastRequestKey       statusRequestKey
+	lastSuccessfulLookup time.Time
```

Comment:
it seems a bit odd to have two different backoff mechanisms, one where the last time is stored and one where the expiry is stored.